draft-dekater-scion-controlplane.md

```diff
@@ -1139,6 +1139,12 @@ Depending on the selection criteria, it may be necessary to keep more candidate
 
 - The *propagation interval* SHOULD be at least "5" (seconds) for intra-ISD beaconing and at least "60" (seconds) for core beaconing.
 
+Note that under certain conditions, to ensure quick connectivity establishment, an AS MAY attempt to forward a PCB more frequently ("fast recovery"):
+
+- if no beacon was successfully sent over the interface in the current propagation interval (e.g. because the corresponding RPC failed)
+
+- or if the AS receives a previously unknown path segment
+
```

Comment:
@jiceatscion @matzf @oncilla do these conditions sound reasonable? How would the AS check that it receives a previously unknown segment, shall we be more precise about this? 

ON a similar note, @tzaeschke 's  comments were: 
>  Is this subject to the "Best 5 PCB" rule? -> If a new path is immediately forwarded, does it count towards th "best 5"? If not, then we are effectively forwarding >5 paths, correct? If yes, then the first 5 paths are always the best until they expire and can be replaced with other paths?

> What means "unknown path"? Does it refer to all links in the segment or just the remote AS? If it is all links in the segment, then there may be many new paths coming in all the time that need to be forwarded immediately, or not?

Reply:
>  How would the AS check that it receives a previously unknown segment, shall we be more precise about this?

I don't think this is accurate. We do not do fast recovery if we receive a previously unknown path segment in all cases.

I think what it really boils down to is the following:

If we have not successfully distributed at least one beacon within the last beaconing interval, be it due to failing RPCs, or because no beacon was available to propagate, we enter the fast recovery mode.

> Is this subject to the "Best 5 PCB" rule? -> If a new path is immediately forwarded, does it count towards th "best 5"? If not, then we are effectively forwarding >5 paths, correct? If yes, then the first 5 paths are always the best until they expire and can be replaced with other paths?

The questions are mute, because we do not immediately beacon if a new path segment is discovered.

Reply:
Ok. so if we want to be both honest and minimalist, we should just give permission to the current implementation do do what it does, without mandating it (since I don't think we really know how good or bad it is). how about:

"
Note that to ensure quick connectivity establishment, an AS MAY attempt to forward a PCB more frequently ("fast recovery") if no attempt is know to have succeeded. Current practice is to increase the frequency of attempts if and only if no beacon has been successfully propagated during the last beaconing interval.
"
?

Reply:
How about this mix of your two wordings? 
```suggestion
Note that to ensure quick connectivity establishment, an AS MAY attempt to forward a PCB more frequently ("fast recovery"). Current practice is to increase the frequency of attempts if no PCB propagation is know to have succeeded within the last propagation interval:

- because the corresponding RPC failed

- or because no beacon was available to propagate
```